cpp/src/io/parquet/experimental/page_index_filter.cu

```diff
@@ -1126,7 +1126,11 @@ thrust::host_vector<bool> aggregate_reader_metadata::compute_data_page_mask(
   //  Search the Fenwick tree to see if there's a surviving row in each page's row range
   auto const num_ranges = static_cast<cudf::size_type>(page_row_offsets.size() - 1);
   rmm::device_uvector<bool> device_data_page_mask(num_ranges, stream, mr);
-  auto page_offsets = cudf::detail::make_device_uvector_async(page_row_offsets, stream, mr);
+  // Use a pinned bounce buffer to avoid pageable h2d copy
```

Comment:
It seems like some of these use cases would be suitable for a (sync/async) pinned vector factory. We have a couple of patterns repeating a lot.

Reply:
Absolutely. It's already implemented in https://github.com/rapidsai/cudf/pull/20120 :)

Reply:
Actually this brings up a larger point. I'm not sure we should use `make_pinned_vector` instead of `make_host_vector` here. Should we switch to `make_pinned_vector` whenever we want to avoid pageable copy, or set the pinned memory use threshold to a non -zero value (~1MB)?

Reply:
I am more inclined towards explicit usage of pinned vector, long term at least.